### ISSUE TITLE
Printer: Incubate update to Print.js

### ISSUE THREAD

**Issue by melloware** [primefaces-extensions/primefaces-extensions#382]
Based on this PF ticket from @stolp https://github.com/primefaces/primefaces/issues/7288

We are going to incubate here using a totally different JS library called Print.js because our current print plugin Canvas is not supported.

**Comment by melloware** [comment]
OK it works!  However when I print a ChartsJS chart I am running into this issue:
https://github.com/crabbly/Print.js/issues/498

It prints the chart but 935 blank pages as well.

**Comment by melloware** [comment]
Was able to reproduce the issue here: https://jsfiddle.net/qgb09xk8/

However if you change from Charts.JS 2.9.3 to 3.2.0 it works OK.  So it may be the way ChartsJS 2.9.3 renders is HTML.

**Comment by jepsar** [comment]
> but 935 blank pages as well

😂

**Comment by melloware** [comment]
Here is a fiddle that works properly with Charts 3.2.0.
https://jsfiddle.net/qgb09xk8/1/

**Comment by melloware** [comment]
To use just add this to your faces-config.xml of your webapp this will override the Primefaces one.
```xml
    <behavior>
        <behavior-id>org.primefaces.component.PrinterBehavior</behavior-id>
        <behavior-class>org.primefaces.extensions.behavior.printer.PrinterBehavior</behavior-class>
    </behavior>
```

**Comment by jepsar** [comment]
Don't we need a note on that in the showcase?

**Comment by melloware** [comment]
Yeah we probably need a note in the Showcase too

**Comment by melloware** [comment]
Showcase example added under ClientBehaviors.

**Comment by stolp** [comment]
I locally built my application against the current extensions-snapshot and added the behaviour.

Sadly, I also get a few hundred more empty pages. Maybe this is related to the embedded ChartJS Version in PrimeFaces 10?

**Comment by melloware** [comment]
Yep  I reported it to Print.js and it will be fixed for PF 11.0 which will use ChartsJS 3.2.0.  But I am hoping Print.js fixes it sooner than later.
